resolver.go

```diff
@@ -36,12 +36,18 @@ const (
 	ptrIPv4domain = ".in-addr.arpa."
 	ptrIPv6domain = ".ip6.arpa."
 	respTTL       = 1800
+	maxExtDNS     = 3 //max number of external servers to try
 )
 
+type extDNSEntry struct {
```

Comment:
Do you really need this ?
You will convert the configured address from string form to net.IP, then when you need to program you will  convert it again back to string from just to store the IPv6 flag.
Are you using anywhere extDNEntry.IP as a net.IP ? Otherwise this seem unnecessary.

I would just check in `NameServer()` with `isIPv6 := net.ParseIP().to4() == nil` in the loop.